src/jira/service.ts

```diff
@@ -0,0 +1,61 @@
+import { IMerjoonProjects, IMerjoonService, IMerjoonTasks, IMerjoonUsers } from "../common/types";
+import { JiraApi } from "./api";
+import { JiraTransformer } from "./transformer";
+import {GetJiraEntity, IJiraIssue, IJiraProject, IJiraUser, JiraApiPath } from "./types"
+
+export class JiraService implements IMerjoonService {
+  constructor(public readonly api: JiraApi, public readonly transformer: JiraTransformer) {}
+
+  protected async* getAllRecordsIterator<T>(path: JiraApiPath, entity: GetJiraEntity) {
+    let currentPage = 0;
+    let isLast = false;
+    const pageSize = Number(this.api.pageSize)
+    do {
+      try {
+        const response: T[] = await this.api[entity](path, {
+            startAt: currentPage * pageSize,
+            maxResults: pageSize
+        });
+        const data: T[] = response
+        yield data;
+        isLast = data.length < pageSize;
+        currentPage++;
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      } catch (e: any) {
```

Comment:
handle error similar to https://github.com/Merjoon/merjoon-core/pull/74/files#diff-b17dd99374a238d783ca7a4e56ee0405f60aeef2f830ad1e61e52dc1fb9a443fR51